server/go/subscription/manager_test.go

```diff
@@ -0,0 +1,108 @@
+package subscription_test
+
+import (
+	"testing"
+
+	"github.com/kubeshop/tracetest/server/go/subscription"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestManagerIsSingleton(t *testing.T) {
+	manager1 := subscription.GetManager()
+	manager2 := subscription.GetManager()
+
+	assert.Equal(t, manager1, manager2, "managers should be the same instance")
```

Comment:
Noob question, when doing an assertion like this, is it done by reference or by something else?

Reply:
In go, unless specified, everything is passed by value. In this case, `manager1` and `manager2` are two potentially different pointers. Meaning that if the singleton is wrong, you would end up with two different pointers. 
So, this is passing `a pointer to a manager`, as the value. Not sure if that makes sense?